erpnext/controllers/status_updater.py

```diff
@@ -281,16 +281,20 @@ def _update_percent_field(self, args, update_modified=True):
 		"""Update percent field in parent transaction"""
 
 		self._update_modified(args, update_modified)
-		
+
 		if args.get('target_parent_field'):
-			frappe.db.sql("""update `tab%(target_parent_dt)s`
-				set %(target_parent_field)s = round(
-					ifnull((select
-						ifnull(sum(if(%(target_ref_field)s > %(target_field)s, abs(%(target_field)s), abs(%(target_ref_field)s))), 0)
-						/ sum(abs(%(target_ref_field)s)) * 100
-					from `tab%(target_dt)s` where parent="%(name)s" having sum(abs(%(target_ref_field)s)) > 0), 0), 6)
-					%(update_modified)s
-				where name='%(name)s'""" % args)
+			frappe.db.sql("""
+			UPDATE `tab%(target_parent_dt)s`
+			SET %(target_parent_field)s = round(ifnull(
+				(
+					SELECT IF(sum(abs(%(target_ref_field)s)) = 0, 100, ifnull(
```

Comment:
Why setting `per_billed` as 100 directly, incase of zero grand total sales / purchase order? In those cases, we are already setting it based on `qty` using `update_billing_status_for_zero_amount_refdoc`?